internal/cmd/platform/update_platform.go

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package platform
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Peripli/service-manager-cli/internal/output"
+	"github.com/Peripli/service-manager-cli/internal/util"
+
+	"github.com/spf13/cobra"
+
+	"github.com/Peripli/service-manager-cli/internal/cmd"
+	"github.com/Peripli/service-manager-cli/pkg/smclient"
+	"github.com/Peripli/service-manager-cli/pkg/types"
+)
+
+// UpdatePlatformCmd wraps the smctl update-platform command
+type UpdatePlatformCmd struct {
+	*cmd.Context
+
+	outputFormat    int
+	name            string
+	updatedPlatform *types.Platform
+}
+
+// NewUpdatePlatformCmd returns new update-platform command with context
+func NewUpdatePlatformCmd(context *cmd.Context) *UpdatePlatformCmd {
+	return &UpdatePlatformCmd{Context: context}
+}
+
+func (upc *UpdatePlatformCmd) buildCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:     "update-platform [name] <json_platform>",
+		Aliases: []string{"up"},
+		Short:   "Updates platform",
+		Long: `Update platform with name.
+Example:
+smctl update-platform platform '{"name": "new-name", "description": "new-description", "type": "new-type"}'`,
```

Comment:
This command needs a -f flag very badly (in a future PR)

Reply:
It already has one. What do you mean?

Reply:
I meant -f as in --file

Reply:
It should be a future PR, which will allow file as input for multiple commands. Will create an issue